src/Nest/CommonAbstractions/SerializationBehavior/StatefulDeserialization/ConcreteTypeConverter.cs

```diff
@@ -94,7 +94,10 @@ internal static class ConcreteTypeConverter
 
 		private static JObject CreateIntermediateJObject(JsonReader reader)
 		{
+			var original = reader.DateParseHandling;
+			reader.DateParseHandling = DateParseHandling.None;
```

Comment:
Great fix. ++ on adding a comment.

Quick question,  where do we use the StatefulDeserialization class? Only for Covariant Search type? Just trying to understand whether this can affect other functionality.